dan_layer/consensus/src/block_validations.rs

```diff
@@ -127,32 +127,40 @@ pub async fn check_quorum_certificate<TConsensusSpec: ConsensusSpec>(
     vote_signing_service: &TConsensusSpec::SignatureService,
     epoch_manager: &TConsensusSpec::EpochManager,
 ) -> Result<(), HotStuffError> {
-    if candidate_block.justify().epoch().as_u64() == 0 {
-        // Ignore genesis block.
+    let qc = candidate_block.justify();
```

Comment:
@sdbondi Please have a look at this logic, I expanded it a bit and it's potentially out of scope of this PR